src/blob.rs

```diff
@@ -410,30 +423,66 @@ impl<'a> BlobObject<'a> {
                 MediaQuality::Worse => WORSE_IMAGE_SIZE,
             };
 
-        self.recode_to_size(context, blob_abs, img_wh).await
+        let new_name = self.recode_to_size(context, blob_abs, img_wh, None).await?;
+        if !new_name.is_empty() {
+            return Err(format_err!(
+                "Internal error: recode_to_size(..., None) shouldn't change the name of the image"
+            )
+            .into());
+        }
+        Ok(())
     }
 
     async fn recode_to_size(
         &self,
         context: &Context,
-        blob_abs: PathBuf,
-        img_wh: u32,
-    ) -> Result<(), BlobError> {
+        mut blob_abs: PathBuf,
+        mut img_wh: u32,
+        max_bytes: Option<usize>,
+    ) -> Result<String, BlobError> {
```

Comment:
I think it's better to explicitly return `Option<String>` and use `None` instead of special empty string value.

Reply:
I have pushed this change to this branch